### PR TITLE
Clawback CAP

### PR DESCRIPTION
**Pull request by dandoney**
CAP proposal to enable a clawback operation for regulated assets to meet regulatory requirements for securities in many jurisdictions.

### CODE REVIEW

File: core/cap-0035.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	union switch (AssetType type)
+    {
+    // ASSET_TYPE_NATIVE is not allowed
+    case ASSET_TYPE_CREDIT_ALPHANUM4:
+        AssetCode4 assetCode4;
+    case ASSET_TYPE_CREDIT_ALPHANUM12:
+        AssetCode12 assetCode12;
+    } asset; // asset code to claw back. asset issuer is implied from source account.  
+	MuxedAccount from;
+	int64 amount;
```
Comment (leighmcculloch) [inline core/cap-0035.md:107]: Minor whitespace inconsistencies here.
```suggestion
struct ClawbackOp 
{
    union switch (AssetType type)
    {
    // ASSET_TYPE_NATIVE is not allowed
    case ASSET_TYPE_CREDIT_ALPHANUM4:
        AssetCode4 assetCode4;
    case ASSET_TYPE_CREDIT_ALPHANUM12:
        AssetCode12 assetCode12;
    } asset; // asset code to claw back. asset issuer is implied from source account.  
    MuxedAccount from;
    int64 amount;
```

File: core/cap-0035.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	int64 amount;	
+}
+ 
+enum ClawbackResultCode
+{
+    // codes considered as "success" for the operation
+    CLAWBACK_SUCCESS = 0, // CLAWBACK successfuly completed
+ 
+    // codes considered as "failure" for the operation
+    CLAWBACK_SRC_NO_TRUST = -1,       // no trust line on source account
```
[the remark this answers is not in the document]
Comment (dandoney) [reply core/cap-0035.md:116]: Good catch. You're right

[the remark this answers is not in the document]
Comment (dandoney) [reply core/cap-0035.md:116]: Agree. Should be malformed if amount<0

File: core/cap-0035.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    CLAWBACK_NOT_ISSUER = -2, // source account is not issuer
```
[the remark this answers is not in the document]
Comment (dandoney) [reply core/cap-0035.md:117]: agreed

File: core/cap-0035.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+traditional blockchain assets (bearer instruments) so that asset owners are aware 
+of rights. The transaction may result in revocation of some or all of the specified 
+assets from the designated account based on the amount provided in the `CLAWBACK` 
+operation. In a future proposal, we will outline an approach by which the same flag 
+can be used to execute a ROLLBACK of a transaction (within a designated time window) 
+in the event of a fraudulent or mistaken transaction.
+
+## Specification
+
+Account changes
```
[the remark this answers is not in the document]
Comment (dandoney) [reply core/cap-0035.md:70]: thanks

**Comment by desinggold** [comment]
Stellar day and time works for you 💞

**Pull request merged**